packages/landingpage/components/getting-started/menuSection.tsx

```diff
@@ -12,10 +14,51 @@ type MenuSectionProps = {
  * @param title The title of the section (that also functions as the id).
  * @param children The content to render within the section.
  */
-function MenuSection({ title, children }: MenuSectionProps) {
+function MenuSection({ title, children, level = 'main' }: MenuSectionProps) {
+  // Sanitize the ID by replacing all non-word characters (this prevents querySelector errors when the id could contain special characters like "Install @inovex.de/elements" in the vue-guide.mdx)
+  const id = title.toCamelCase().replace(/[^\w-]/g, '-');
+  const isMainSection = level === 'main';
+
+  const handleMenuSectionClick = (
+    event: React.MouseEvent<HTMLHeadingElement>,
+  ) => {
+    event.preventDefault();
+    const targetElement = document.getElementById(id);
+    if (!targetElement) return;
+
+    const headerOffset = 80;
+    const elementPosition = targetElement.getBoundingClientRect().top;
+    const offsetPosition = elementPosition + window.pageYOffset - headerOffset;
+
```

Comment:
These lines are Duplicates. Maybe we can move this logic into a custom hook like we did in [useStorybookUr](https://github.com/inovex/elements/blob/master/packages/landingpage/utils/hooks/useStorybookUrl.ts)l / [useStorybookSyncer](https://github.com/inovex/elements/blob/master/packages/landingpage/utils/hooks/useStorybookUrlSyncer.ts)